lib/filewatch/watch.rb

```diff
@@ -126,77 +124,136 @@ def inode(path,stat)
     #   :create - file is created (new file after initial globs, start at 0)
     #   :modify - file is modified (size increases)
     #   :delete - file is deleted
-    public
     def each(&block)
       synchronized do
+        return if @files.empty?
+
+        file_deleteable = []
+        # look at the closed to see if its changed
+
+        @files.values.select {|wf| wf.closed? }.each do |watched_file|
```

Comment:
If you're worried about object churn here, you can use @files.select {|_, wf| wf.closed? }.each do |_, wf| ... end since Hash#select returns an enumerable.